ignite/handlers/param_scheduler.py

```diff
@@ -1008,16 +1017,17 @@ def print_lr():
             param_group_warmup_end_value = warmup_end_value
 
         milestones_values = [(0, warmup_start_value), (warmup_duration - 1, param_group_warmup_end_value)]
-
         if isinstance(lr_scheduler, _LRScheduler):
             init_lr = param_group["lr"]
 
             if init_lr != param_group_warmup_end_value:
```

Comment:
@yuta0821 I wonder if this part of code shouldn't be changed to avoid extra step with initial optimizer's lr ?

Following your examples (example1 and example2): https://github.com/pytorch/ignite/pull/2496#discussion_r815573334

Example 1: 
```
optimizer = torch.optim.SGD([tensor], lr=0.01)
torch_lr_scheduler = StepLR(optimizer, step_size=3, gamma=0.1)
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=0.0,
                                            warmup_end_value=0.1,
                                            warmup_duration=3)
```

What would be an expected output regardless the implementation ? IMO, something like:
```
0.0     # iter0, warmup
0.05    # iter1, warmup
0.1     # iter2, warmup
0.01    # iter3, torch scheduler
0.01    # iter4, torch scheduler
0.01    # iter4, torch scheduler
0.001
```

Example 2: 
```
optimizer = torch.optim.SGD([tensor], lr=0.1)
torch_lr_scheduler = StepLR(optimizer, step_size=3, gamma=0.1)
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=0.0,
                                            warmup_end_value=0.1,
                                            warmup_duration=3)
```

What would be an expected output regardless the implementation ? IMO, something like:
```
0.0     # iter0, warmup
0.05    # iter1, warmup
0.1     # iter2, warmup
0.01    # iter3, torch scheduler
0.01    # iter4, torch scheduler
0.01    # iter4, torch scheduler
0.001
```

Reply:
Sorry, I wanted to say "this part of code SHOULD be changed" and not "shouldn't".

Reply:
Yes, and is this overlap due to : 
```python
if init_lr != param_group_warmup_end_value:
    milestones_values.append((warmup_duration, init_lr))
```

Reply:
Yes, it is.
However, if this part of code is removed, the following easy test doesn't work well.

```python
optimizer = torch.optim.SGD([tensor], lr=0.01)
torch_lr_scheduler = StepLR(optimizer, step_size=3, gamma=0.1)
scheduler = create_lr_scheduler_with_warmup(torch_lr_scheduler,
                                            warmup_start_value=0.0,
                                            warmup_end_value=0.1,
                                            warmup_duration=3)

```

```
0.0      # iter0, warmup
0.05     # iter1, warmup
0.1      # iter2, warmup
0.1      # iter3, torch scheduler <- this result is not good
0.1      # iter4, torch scheduler
0.1      # iter4, torch scheduler
0.01     # iter5, torch scheduler
```

Reply:
@yuta0821 thanks for checking that ! OK, let's keep this part of code.